plugins/auth-backend/src/providers/types.ts

```diff
@@ -112,6 +112,14 @@ export interface AuthProviderRouteHandlers {
   logout?(req: express.Request, res: express.Response): Promise<void>;
 }
 
+export type IdentityResolver = (
```

Comment:
Wondering if we can call this experimental for now. We have plenty of plans here, as freben mentioned, but don't wanna have to block this PR on getting the interface perfect.

Reply:
Yes, totally - this should be replaced with what is decided in #4089  - this is just a placeholder

Reply:
I changed the type name here to ExperimentalIdentityResolver and added a comment in the type documentation calling out that it almost certain to change in the future.